main.c

```diff
@@ -64,6 +68,8 @@ accelGenericInterrupt_t accelInterrupt1 = {
 };
 
 static uint8_t flashReadBuffer[FLASH_READ_BUFFER_SIZE] = {0};
+static int16_t* micData;
+int16_t test_buffer[PDM_BUFFER_LENGTH] = {0};

```

Comment:
this looks like an artifact of code you have commented out, can you remove this?